data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/BeaconRestApi.java

```diff
@@ -52,11 +53,12 @@ private void initialise(
       ChainStorageClient chainStorageClient,
       P2PNetwork<?> p2pNetwork,
       HistoricalChainData historicalChainData,
+      CombinedChainDataClient combinedChainDataClient,
       final int requestedPortNumber) {
     app.server().setServerPort(requestedPortNumber);
 
     addNodeHandlers(chainStorageClient);
-    addBeaconHandlers(chainStorageClient, historicalChainData);
+    addBeaconHandlers(chainStorageClient, historicalChainData, combinedChainDataClient);
```

Comment:
As something to work towards, it makes sense to me that the REST-APIs would only have a `CombinedChainDataClient`. It contains both current and historical data so just a matter of ensuring it has the right methods to delegate through.  Then the REST APIs never need to worry about whether something has been finalised or not as they always use the same API.

Reply:
Yep I think that's where I'm heading with this change... it makes sense that the interface shouldn't need to understand the storage scheme etc. and combined client gives us that abstraction.